examples/calculator/lib/main.dart

```diff
@@ -0,0 +1,261 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'calc_expression.dart';
+
+// A calculator application.
+void main() {
+  runApp(new MaterialApp(
+      title: 'Calculator',
+      theme: new ThemeData(primarySwatch: Colors.blue),
+      home: new Calculator()));
+}
+
+class Calculator extends StatefulWidget {
+  Calculator({Key key}) : super(key: key);
+
+  @override
+  _CalculatorState createState() => new _CalculatorState();
+}
+
+class _CalculatorState extends State<Calculator> {
+  // As the user taps keys we update the current |_expression| and we also
+  // keep a stack of previous expressions so we can return to earlier states
+  // when the user hits the DEL key.
+  final List<CalcExpression> _expressionStack = <CalcExpression>[];
+  CalcExpression _expression = new CalcExpression.Empty();
+
+  // Make |expression| the current expression and push the previous current
+  // expression onto the stack.
+  void pushExpression(CalcExpression expression) {
+    _expressionStack.add(_expression);
+    _expression = expression;
+  }
+
+  // Pop the top expression off of the stack and make it the current expression.
+  void popCalcExpression() {
+    if (_expressionStack.length > 0) {
+      _expression = _expressionStack.removeLast();
+    } else {
+      _expression = new CalcExpression.Empty();
+    }
+  }
+
+  // Set |resultExpression| to the currrent expression and clear the stack.
+  void setResult(CalcExpression resultExpression) {
+    _expressionStack.clear();
+    _expression = resultExpression;
+  }
+
+  void onNumberTap(int n) {
```

Comment:
nit: we name properties onFoo, functions that are assigned to those properties are handleFoo.